### DISCUSSION TITLE
Artifacts not populating in side panel

### DISCUSSION THREAD

**Discussion by thistleknot** [open-webui/open-webui#6111] [Q&A] [answered]
sorry for being a newb, but how do I setup open-webui to produce artifacts?  Do I need a custom prompt?  Do I need a specific llm?  I saw a youtube video and it showed qwen-coder 2.5 7b working with artifacts, so I should be able to do this with a local llm.  I have a very long system prompt I got online... when not having a system prompt also produced no artifacts... I see the artifacts pane, but it never populates.

![image](https://github.com/user-attachments/assets/18dcf42a-9bc4-4257-b494-1f24f2589d84)

**Comment by silentoplayz** [comment, marked as answer]
A special system prompt is not necessary. When an LLM provides a valid HTML response, Open WebUI will automatically detect the HTML contents and render them in the right-hand `Artifacts` sidebar, allowing for live rendering and display. Currently, artifacts within Open WebUI are able to render out web view code (CSS, HTML, and JavaScript) in combination.
